src/pretix/static/pretixcontrol/js/ui/popover.js

```diff
@@ -0,0 +1,42 @@
+$(function () {
+    var basePath = JSON.parse(document.getElementById('base_path').textContent);
+    var currentPath = window.location.pathname;
+    var queryString = window.location.search;
+    var logoutPath = `/control/logout?back=${encodeURIComponent(currentPath)}%3F${encodeURIComponent(queryString)}`;
```

Comment:
Please re-implement the part of `encodeURIComponent`, `decodeURIComponent` with [`URLSearchParams`](https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams). It is difficult to review this code, it looks easy to produce bugs here.